multiaccounts/settings/database.go

```diff
@@ -0,0 +1,299 @@
+		&s.Address,
+		&s.AnonMetricsShouldSend,
+		&s.ChaosMode,
+		&s.Currency,
+		&s.CurrentNetwork,
+		&s.CustomBootnodes,
+		&s.CustomBootnodesEnabled,
+		&s.DappsAddress,
+		&s.EIP1581Address,
+		&s.Fleet,
+		&s.HideHomeTooltip,
+		&s.InstallationID,
+		&s.KeyUID,
+		&s.KeycardInstanceUID,
+		&s.KeycardPAiredOn,
+		&s.KeycardPairing,
+		&s.LastUpdated,
+		&s.LatestDerivedPath,
+		&s.LinkPreviewRequestEnabled,
+		&s.LinkPreviewsEnabledSites,
+		&s.LogLevel,
+		&s.Mnemonic,
+		&s.Name,
+		&s.Networks,
+		&s.NotificationsEnabled,
+		&s.PushNotificationsServerEnabled,
+		&s.PushNotificationsFromContactsOnly,
+		&s.RemotePushNotificationsEnabled,
+		&s.SendPushNotifications,
+		&s.PushNotificationsBlockMentions,
+		&s.PhotoPath,
+		&s.PinnedMailserver,
+		&s.PreferredName,
+		&s.PreviewPrivacy,
+		&s.PublicKey,
+		&s.RememberSyncingChoice,
+		&s.SigningPhrase,
+		&s.StickerPacksInstalled,
+		&s.StickerPacksPending,
+		&s.StickersRecentStickers,
+		&s.SyncingOnMobileNetwork,
+		&s.DefaultSyncPeriod,
+		&s.UseMailservers,
+		&s.MessagesFromContactsOnly,
+		&s.Usernames,
+		&s.Appearance,
+		&s.ProfilePicturesShowTo,
+		&s.ProfilePicturesVisibility,
+		&s.WalletRootAddress,
+		&s.WalletSetUpPassed,
+		&s.WalletVisibleTokens,
+		&s.WakuBloomFilterMode,
+		&s.WebViewAllowPermissionRequests,
+		&sqlite.JSONBlob{Data: &s.CurrentUserStatus},
+		&s.SendStatusUpdates,
+		&sqlite.JSONBlob{Data: &s.GifRecents},
+		&sqlite.JSONBlob{Data: &s.GifFavorites},
+		&s.OpenseaEnabled,
+		&s.LastBackup,
+		&s.BackupEnabled,
+		&s.TelemetryServerURL,
+		&s.AutoMessageEnabled,
+	)
+
+	return s, err
+}
+
+func (db *Database) GetSettingLastSynced(column string) (uint64, error) {
+	var result uint64
+
+	query := "SELECT %s FROM settings_sync_clock WHERE synthetic_id = 'id'"
+	query = fmt.Sprintf(query, column)
+
+	err := db.db.QueryRow(query).Scan(&result)
+	if err != nil {
+		return 0, err
+	}
+
+	return result, nil
+}
+
+func (db *Database) SetSettingLastSynced(column string, clock uint64) error {
```

Comment:
What do you think of changing this function so it's not exported, to avoid the possibility of it being called by another package, and having the package needing to know a DB column name (which IMO it's an implementation detail that should only be known by this package).

```suggestion
func (db *Database) setSettingLastSynced(column string, clock uint64) error {
```

If you still prefer to have this function exported, maybe It can be changed so it receives a SettingField instead of the column name? and the the column value could be extracted with `theSetting.GetDbName()`

Reply:
I agree with this also. I think I'll implement both your suggestions because I'd like to make it hard to mess with the sync clocks be accident.